blackbox/docs/general/ddl/data-types.rst

```diff
@@ -316,6 +317,114 @@ converted to UTC without considering the time zone indication.
     date time types. That means date type columns must always be declared
     beforehand.
 
+.. _interval_data_type:
+
+Interval
+--------
+
+
+.. _interval-literal:
+
+Interval literal
+................
+
+An interval literal represents a span of time and is either
+a :ref:`year-month-literal` or a :ref:`day-time-literal` literal. The generic
+literal definition defined as following
+
+::
+
+    <interval_literal> ::=
+        INTERVAL [ <sign> ] <string_literal> <interval_qualifier>
+
+    <interval_qualifier> ::=
+        <start_field> [ TO <end_field>]
+
+    <start_field> ::= <datetime_field>
+    <end_field> ::= <datetime_field>
+
+    <datetime_field> ::=
+          YEAR
+        | MONTH
+        | DAY
+        | HOUR
+        | MINUTE
+        | SECOND
+
+.. _year-month-literal:
+
+year-month
+^^^^^^^^^^
+
+A ``year-month`` literal includes either ``YEAR``, ``MONTH`` or a contiguous
+subset of these fields.
+
+::
+
+    <year_month_literal> ::=
+        INTERVAL [ {+ | -} ]'yy' <interval_qualifier> |
+        INTERVAL [ {+ | -} ]'[ yy- ] mm' <interval_qualifier>
+
+For example
+
+::
+
+    INTERVAL '01' YEAR              - 1 year
+    INTERVAL '02' MONTH             - 2 months
+    INTERVAL '01-02' YEAR TO MONTH  - 1 year and 2 months
+
+.. _day-time-literal:
+
+day-time
+^^^^^^^^
+
+A ``day-time`` literal includes either ``DAY``, ``HOUR``, ``MINUTE``,
+``SECOND`` or a contiguous subset of these fields.
+
+When using ``SECOND``, it is possible to define more digits representing
+a number of fractions of a seconds with ``.nn``. The allowed fractional
+seconds precision of ``SECOND`` ranges from 0 to 6 digits.
+
+::
+
+    <day_time_literal> ::=
+        INTERVAL [ {+ | -} ]'dd [ <space> hh [ :mm [ :ss ]]]' <interval_qualifier>
+        INTERVAL [ {+ | -} ]'hh [ :mm [ :ss [ .nn ]]]' <interval_qualifier>
+        INTERVAL [ {+ | -} ]'mm [ :ss [ .nn ]]' <interval_qualifier>
+        INTERVAL [ {+ | -} ]'ss [ .nn ]' <interval_qualifier>
+
+For example
+
+::
+
+    INTERVAL '100.123' SECOND         - 100.123 seconds
+    INTERVAL '40 23' DAY TO HOUR      - 40 days and 23 hours
+    INTERVAL '10 23:10' DAY TO MINUTE - 10 days, 23 hours and 10 minutes
+
+.. _temporal-arithmetic:
+
+Temporal arithmetic
+-------------------
+
+The following table specifies the declared types of
+:ref:`arithmetic <arithmetic>` expressions that involves temporal operands.
+
++---------------+----------+---------------+
```

Comment:
We are a bit more lenient with temporal arithmetic operations, it is just the documented subset and what sql standard defines.